pkg/kernelrelease/kernelrelease.go

```diff
@@ -1,25 +1,53 @@
 package kernelrelease
 
 import (
+	"fmt"
 	"log"
 	"regexp"
 	"strconv"
+	"strings"
 )
 
 var (
 	kernelVersionPattern = regexp.MustCompile(`(?P<fullversion>^(?P<version>0|[1-9]\d*)\.(?P<patchlevel>0|[1-9]\d*)\.(?P<sublevel>0|[1-9]\d*))(?P<fullextraversion>-(?P<extraversion>0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-_]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$`)
 )
 
+// Architectures is a Map [Architecture] -> non-deb-ArchitectureString
+type Architectures map[Architecture]string
+
+// SupportedArchs enforces the duality of architecture->non-deb one when adding a new one
+var SupportedArchs = Architectures{
```

Comment:
The idea is that adding support for a new architecture is as easy as adding a new entry here.
Note: i made practically impossible to forget to add also the Non-deb name (previously handled in a `ToNonDeb()` method switch case statement).